### PR TITLE
Add commented arguments and new methods to FileSystem for Transactions

### PR DESCRIPTION
**Pull request by samikama**
This PR is a continuation of series of PRs introducing Transactions API to FileSystems. In this PR new Transaction API methods have been added to FileSystem base class and existing method signatures are changed to add Transaction Token arguments in commented form. Previous PRs, #41431, #41433, #41434, #41462 and #41463 and this PR merged, I will open another PR to uncomment the extra arguments and enable new API.

### CODE REVIEW

File: tensorflow/core/platform/file_system.cc [hunk 1]
```diff
@@ -436,4 +436,13 @@ string FileSystem::CreateURI(StringPiece scheme, StringPiece host,
   return strings::StrCat(scheme, "://", host, path);
 }
 
+string FileSystem::DecodeTransaction(const TransactionToken* token){
+  if(token){
+    std::stringstream oss;
+    oss<<"Token= "<<token->token<<", Owner="<<token->owner;
+    return oss.str();
+  }
```
Comment (mihaimaruseac) [inline tensorflow/core/platform/file_system.cc:444]: Let's use `StrCat` (https://cs.opensource.google/tensorflow/tensorflow/+/master:tensorflow/core/platform/strcat.h;l=147-176;drc=16b5232a803abcb55a30fa8326edbe279aca2a1e)

**Review by mihaimaruseac — CHANGES_REQUESTED** [review]

File: tensorflow/core/platform/file_system.cc (hunk 1 above)
Comment (samikama) [reply tensorflow/core/platform/file_system.cc:444]: Hi, @mihaimaruseac,

This is what I intended to use it first but AFAIK StrCat can not handle void* without some casting. Am I  wrong?

Comment (mihaimaruseac) [reply tensorflow/core/platform/file_system.cc:444]: Let's add the needed cast/`ToString` method.

Also, this should return `std::string`, not `string` (will probably trip the linter upon import, even if it compiles now)

Comment (samikama) [reply tensorflow/core/platform/file_system.cc:446]: @mihaimaruseac 
I added a TODO message for switching to StrCat when void* is supported and changed the signature to std::string.

File: tensorflow/core/platform/file_system.cc [hunk 2]
```diff
@@ -436,4 +436,15 @@ string FileSystem::CreateURI(StringPiece scheme, StringPiece host,
   return strings::StrCat(scheme, "://", host, path);
 }
 
+std::string FileSystem::DecodeTransaction(const TransactionToken* token){
+
+// TODO(sami): Switch using StrCat when void* is supported
+  if(token){
+    std::stringstream oss;
+    oss<<"Token= "<<token->token<<", Owner="<<token->owner;
```
Comment (mihaimaruseac) [inline tensorflow/core/platform/file_system.cc:444]:
```suggestion
    oss << "Token= " << token->token << ", Owner=" << token->owner;
```

**Review by mihaimaruseac — CHANGES_REQUESTED** [review]

**Review by mihaimaruseac — APPROVED** [review]

**Pull request merged**